File: 0202-happy-number/0202-happy-number.java
class Solution {
    public boolean isHappy(int n) {
        
        HashSet<Integer> found = new HashSet<>();

        while (n != 0 && ! found.contains(n)) {
            found.add(n);

            int Sum = 0;
            while (n > 0) {
                int digit = n % 10;
                Sum += digit * digit;
                n = n / 10;
            }
            n = Sum;
        }
        return n == 1;
    }
}

File: 0202-happy-number/0202-happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        found = set()
        while n != 0 and n not in found:
            found.add(n)
        
            Sum = 0
            while n > 0:
                digit = n % 10
                Sum += digit * digit
                n = n // 10
            n = Sum

        return  n == 1